src/Storages/MergeTree/MergeTreeIndexConditionText.cpp

```diff
@@ -62,21 +63,33 @@ MergeTreeIndexConditionText::MergeTreeIndexConditionText(
     : WithContext(context_)
     , header(index_sample_block)
     , token_extractor(token_extractor_)
-    , use_bloom_filter(context_->getSettingsRef()[Setting::text_index_use_bloom_filter])
     , preprocessor(preprocessor_)
-    , use_dictionary_block_cache(context_->getSettingsRef()[Setting::use_text_index_dictionary_cache])
-    , dictionary_block_cache(context_->getTextIndexDictionaryBlockCache())
-    , use_header_cache(context_->getSettingsRef()[Setting::use_text_index_header_cache])
-    , header_cache(context_->getTextIndexHeaderCache())
-    , use_postings_cache(context_->getSettingsRef()[Setting::use_text_index_postings_cache])
-    , postings_cache(context_->getTextIndexPostingsCache())
 {
     if (!predicate)
     {
         rpn.emplace_back(RPNElement::FUNCTION_UNKNOWN);
         return;
     }
 
+    const auto & settings = context_->getSettingsRef();
```

Comment:
Assuming the user disabled caching, we still create query-local "opportunistic" caches. This is unusual logic. I think we should add a comment that explains the motivation (probably it is performance).